src/components/common/ColonyTotalFunds/ColonyTotalFundsSelectedToken.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useMemo, useState, useEffect } from 'react';
+import { defineMessages } from 'react-intl';
+
+import Icon from '~shared/Icon';
+import Numeral from '~shared/Numeral';
+// import { MiniSpinnerLoader } from '~shared/Preloaders';
+// import { Colony, useTokenBalancesForDomainsQuery } from '~data/index';
+import { Address } from '~types/index';
+import { getTokenDecimalsWithFallback } from '~utils/tokens';
+// import { COLONY_TOTAL_BALANCE_DOMAIN_ID } from '~constants';
+import { useColonyContext } from '~hooks';
+import { notNull } from '~utils/arrays';
+
+import ColonyTotalFundsPopover from './ColonyTotalFundsPopover';
+
+import styles from './ColonyTotalFunds.css';
+import IconTooltip from '~shared/IconTooltip';
+
+const MSG = defineMessages({
+  loadingData: {
+    id: 'dashboard.ColonyTotalFundsSelectedToken.loadingData',
+    defaultMessage: 'Loading token information...',
+  },
+  tokenSelect: {
+    id: 'dashboard.ColonyTotalFundsSelectedToken.tokenSelect',
+    defaultMessage: 'Select Tokens',
+  },
+});
+
+type Props = {
+  children?: React.ReactChild;
+};
+
+const displayName = 'dashboard.ColonyTotalFundsSelectedToken';
+
+const ColonyTotalFundsSelectedToken = ({ children }: Props) => {
+  const { colony } = useColonyContext();
+  const { tokens, nativeToken, status } = colony || {};
+  const { tokenAddress: nativeTokenAddress } = nativeToken || {};
+
+  // const { data, loading: isLoadingTokenBalances } =
+  //   useTokenBalancesForDomainsQuery({
+  //     variables: {
+  //       colonyAddress: colony?.colonyAddress,
+  //       domainIds: [COLONY_TOTAL_BALANCE_DOMAIN_ID],
+  //       tokenAddresses: colonyTokens.map(({ address }) => address),
+  //     },
+  //   });
+
+  const [currentTokenAddress, setCurrentTokenAddress] = useState<Address>();
+
+  useEffect(() => {
+    if (!nativeTokenAddress) {
+      return;
+    }
+
+    setCurrentTokenAddress(nativeTokenAddress);
+  }, [nativeTokenAddress]);
+
+  const currentToken = useMemo(() => {
+    if (tokens) {
+      return tokens.items.find(
```

Comment:
This could be wrapped into a small util `getCurrentToken` and moved outside the component. Then we could remove the `useMemo`